Core/ALife.Core/Utility/Maths/ExtraMaths.tt

```diff
@@ -0,0 +1,144 @@
+﻿<#@ template debug="false" hostspecific="false" language="C#" #>
```

Comment:
Why Template file for this?

Reply:
We can't have generic math in the version of .net we're at and this way the ExtraMath methods (which should realistically be available for any type) are available for any type. Once we're on .net7 or 8 this templated version can go away and we can have a single generic version of each method

Reply:
(i.e. I got tired of copying/pasting the same method each time I needed to update it for a new numeric type).